packages/data-stores/src/auth/actions.ts

```diff
@@ -126,48 +127,94 @@ export function createActions( {
 		const username = yield select( STORE_KEY, 'getUsernameOrEmail' );
 
 		try {
-			const loginResponse = yield fetchAndParse(
-				// TODO Wrap this in `localizeUrl` from lib/i18n-utils
-				'https://wordpress.com/wp-login.php?action=login-endpoint',
-				{
-					credentials: 'include',
-					method: 'POST',
-					headers: {
-						Accept: 'application/json',
-						'Content-Type': 'application/x-www-form-urlencoded',
-					},
-					body: stringify( {
-						remember_me: true,
-						username,
-						password,
-						client_id,
-						client_secret,
-					} ),
-				}
-			);
+			let loginResponse = yield* wpLogin( 'login-endpoint', {
+				remember_me: true,
+				username,
+				password,
+			} );
+
+			if ( ! loginResponse.success ) {
+				return receiveWpLoginFailed( loginResponse );
+			}
+
+			if ( loginResponse.data.two_step_notification_sent ) {
+				yield receiveWpLogin( loginResponse );
 
-			if ( loginResponse.ok && loginResponse.body.success ) {
-				if ( loadCookiesAfterLogin ) {
-					yield reloadProxy();
+				if ( loginResponse.data.two_step_notification_sent === 'push' ) {
+					loginResponse = yield* handlePush2fa( loginResponse );
```

Comment:
Yea, I think it would be better to use two different `const`'s here for the original `loginRepsonse` and the 2fa push response

Reply:
My rationale for reusing the variable here was so I could easily pass it to `handleSuccessfulLogin()` regardless of what branches we go down.
It's certainly doable without reusing the variable, but have you got an idea for doing this in a relatively clean way?

Reply:
After a refactor I no longer need to have a mutable `loginResponse` variable 👍